src/object.d

```diff
@@ -1736,6 +1736,42 @@ class TypeInfo_Class : TypeInfo
         }
         return o;
     }
+
+   /**
+    * Returns true if the class described by this TypeInfo_Class derives from
+    * the interface described by `parent`.
+    */
+    bool isDerivedFrom(scope const TypeInfo_Interface parent) const @nogc nothrow pure @safe
+    {
+        return parent !is null && interfaceConvertibleFrom(parent.info, this);
+    }
+
+   /**
+    * Returns true if the class described by this TypeInfo_Class describes a class that
+    * derives from the interface described by this TypeInfo_Interface.
+    */
+    bool isDerivedFrom(scope const TypeInfo_Class parent) const @nogc nothrow pure @safe
+    {
+        if (parent is null)
+            return false;
+        if (parent is this || parent.name == this.name)
+            return true;
+        return this.base !is null && this.base.isDerivedFrom(parent);
+    }
+
+    private static bool interfaceConvertibleFrom(scope const TypeInfo_Class interfaceInfo,
```

Comment:
I also recommend DDoc headers on private functions because it makes it easier for those browsing the code with a modern editor to see the documentation at usage sights.